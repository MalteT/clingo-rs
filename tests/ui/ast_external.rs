use clingo::*;

fn main() {
    let sym = Symbol::create_id("test", true).unwrap();
    let term = ast::Term::from(sym);
    let ext = ast::External::new(term, &[]);
    drop(term);
    let _end = ext;
}