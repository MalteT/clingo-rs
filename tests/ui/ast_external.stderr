error[E0382]: use of moved value: `term`
 --> $DIR/ast_external.rs:7:10
  |
5 |     let term = ast::Term::from(sym);
  |         ---- move occurs because `term` has type `clingo::ast::Term<'_>`, which does not implement the `Copy` trait
6 |     let ext = ast::External::new(term, &[]);
  |                                  ---- value moved here
7 |     drop(term);
  |          ^^^^ value used here after move

For more information about this error, try `rustc --explain E0382`.
